hermes-mock/src/main/java/pl/allegro/tech/hermes/mock/HermesMock.java

```diff
@@ -0,0 +1,210 @@
+                    .willReturn(responseDefinitionBuilder
+                            .withHeader("Content-Type", "avro/binary")
+                            .withHeader("Hermes-Message-Id", UUID.randomUUID().toString())
+                    )
+            );
+        } else {
+            wireMockServer.stubFor(get(urlPathMatching("/topics/" + topicName))
+                    .willReturn(responseDefinitionBuilder
+                            .withHeader("Content-Type", "application/json")
+                            .withHeader("Hermes-Message-Id", UUID.randomUUID().toString())
+                    )
+            );
+        }
+    }
+
+    public void expectSingleMessageOnTopic(String topicName) {
+        expectMessagesOnTopic(1, topicName);
+    }
+
+    public void expectMessagesOnTopic(int count, String topicName) {
+        try {
+            await().atMost(awaitSeconds, SECONDS).until(() ->
+                    wireMockServer.verify(count, postRequestedFor(urlEqualTo("/topics/" + topicName)))
+            );
+        } catch (Exception ex) {
+            throw new HermesMockException("Hermes mock did not receive " + count + " messages. " + ex.getMessage());
+        }
+    }
+
+    public <T> void expectSingleMessageOnTopicAs(String topicName, Class<T> clazz) {
+        expectMessagesOnTopicAs(1, topicName, clazz);
+    }
+
+    public <T> void expectMessagesOnTopicAs(int count, String topicName, Class<T> clazz) {
+        try {
+            await().atMost(awaitSeconds, SECONDS).until(() ->
+                    wireMockServer.verify(count, postRequestedFor(urlEqualTo("/topics/" + topicName)))
+            );
+        } catch (Exception ex) {
+            throw new HermesMockException("Hermes mock did not received " + count + " messages. " + ex.getMessage());
+        }
+        List<T> allMessages = getAllMessagesAs(topicName, clazz);
+        if (allMessages.size() != count) {
+            throw new HermesMockException("Hermes mock did not received " + count + " messages, got " + allMessages.size());
+        }
+    }
+
+    public void expectSingleAvroMessageOnTopic(String topicName, Schema schema) {
+        expectAvroMessagesOnTopic(1, topicName, schema);
+    }
+
+    public void expectAvroMessagesOnTopic(int count, String topicName, Schema schema) {
+        try {
+            await().atMost(awaitSeconds, SECONDS).until(() ->
+                    wireMockServer.verify(count, postRequestedFor(urlEqualTo("/topics/" + topicName)))
+            );
+        } catch (Exception ex) {
+            throw new HermesMockException("Hermes mock did not received " + count + " messages. " + ex.getMessage());
+        }
+        if (getAllAvroMessagesAs(topicName, schema).size() != count) {
+            throw new HermesMockException("Hermes mock did not received " + count + " messages, got " + getAllAvroMessagesAs(topicName, schema).size());
+        }
+    }
+
+    public List<LoggedRequest> getAllRequests() {
```

Comment:
Again, I would not respond with Wiremock classes. `LoggedRequest` contains a lot of fields that make no sense in hermes context, like `cookies` or `isBrowserProxyRequest`. Let's respond only with what the user needs to debug his tests.